lib/portlayer/attach/connector.go

```diff
@@ -168,16 +168,24 @@ func (c *Connector) processIncoming(conn net.Conn) {
 		// 2 or more bytes it will just wait, so client should timeout.
 		// However, if timeout is too short, client will flood server with Syn requests.
 		ctx, cancel := context.WithTimeout(context.TODO(), time.Second)
-		if err = serial.HandshakeClient(ctx, conn, c.debug); err == nil {
+		deadline, ok := ctx.Deadline()
+		if ok {
+			conn.SetReadDeadline(deadline)
+		}
+		if err = serial.HandshakeClient(conn, c.debug); err == nil {
```

Comment:
FYI, removing ctx from the signature is going to make it harder to add trace.Operation based logging when it comes.